server/Dockerfile

```diff
@@ -47,82 +50,68 @@ FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim" as php-build
-
-    # install STIX2.0 library to support STIX 2.0 export:
-    # Original Requirements has a bunch of non-required pacakges, force it to only grab wheels for deps from setup.py
-    RUN git clone --depth 1 https://github.com/MISP/cti-python-stix2.git; \
-        cd cti-python-stix2 || exit; python3 setup.py bdist_wheel -d /wheels; \
-        echo "-e ." > requirements.txt; pip3 wheel -r requirements.txt --no-cache-dir -w /wheels/
-
-    # Install the new build tool
-    RUN pip3 install build
-
-    # install PyMISP
-    RUN git clone --depth 1 https://github.com/MISP/PyMISP.git; \
-        cd PyMISP || exit; python3 -m build --wheel --outdir /wheels
-
-    # install pydeep2 (drop-in replacement for pydeep)
-    RUN git clone --depth 1 https://github.com/JakubOnderka/pydeep.git; \
-        cd pydeep || exit; python3 setup.py bdist_wheel -d /wheels
-
-    # Grab other modules we need
-    RUN pip3 wheel --no-cache-dir -w /wheels/ plyara pyzmq redis python-magic lief>=0.13.1
-
-    # Remove extra packages due to incompatible requirements.txt files
-    WORKDIR /wheels
-    RUN find . -name "Sphinx*" | tee /dev/stderr | grep -v "Sphinx-1.5.5" | xargs rm -f
 
+    RUN apt-get update; apt-get install -y --no-install-recommends python3-pip git && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+    ARG MISP_TAG
+    ARG MISP_COMMIT
+    # Download MISP using git in the /var/www/ directory. Remove unnecessary items.
+    RUN <<-EOF
+        mkdir /wheels
+        if [ ! -z "${MISP_COMMIT}" ]; then
+            git clone https://github.com/MISP/MISP.git /var/www/MISP && cd /var/www/MISP && git checkout "${MISP_COMMIT}"
+        else
+            git clone --branch "${MISP_TAG}" --depth 1 https://github.com/MISP/MISP.git /var/www/MISP
+        fi
+
+        cd /var/www/MISP || exit; git submodule update --init --recursive .
+
+        # Until MISP project specifies its required PyMISP version, we grab this from the submodule
+        pymispversion=$(head -n1 PyMISP/pymisp/__init__.py | awk '{print $3}' | tr -d \')
+
+        if [ -n "$pymispversion" ]; then
+            sed -i "s/pymisp$/pymisp==$pymispversion/" requirements.txt
+            cat requirements.txt
+        fi
+        # End of pymisp version fixing hack
+
+        # Remove modules only used during MISP build
+        set -- 'coveralls' 'codecov' 'requests-mock' 'nose' 'pip'
+        for mod in "$@"; do
+            sed "/${mod}/d" -i /var/www/MISP/requirements.txt
+        done;
+
+        # Add additional dependencies (container specific)
+        set -- "redis${PYPI_REDIS_VERSION}" "lief${PYPI_LIEF_VERSION}" "pydeep2${PYPI_PYDEEP2_VERSION}" "python-magic${PYPI_PYTHON_MAGIC_VERSION}" "misp-lib-stix2${PYPI_MISP_LIB_STIX2_VERSION}" "maec${PYPI_MAEC_VERSION}" "mixbox${PYPI_MIXBOX_VERSION}" "cybox${PYPI_CYBOX_VERSION}"
+        for mod in "$@"; do
+            grep -qxF ${mod} /var/www/MISP/requirements.txt || echo ${mod} >> /var/www/MISP/requirements.txt
```

Comment:
I am not sure this works as expected.

Imagine the case where `requirements.txt` has a dependency `dep=0.1`, and the Dockerfile has `dep=0.5`.
Both would be added. I guess the grep should work only on the name of the dependency.

Reply:
Actually we should give priority to our versions IFF they are defined to non-null strings.

Reply:
OK I've [implemented this logic](https://github.com/ostefano/docker-misp/pull/27/commits/a0e640ef881e66cd6b44de2f3ca130ba1def79c8) now.

For each module we want to add,

- If it's missing in requirements.txt, just add it, optionally with version specified
- If it's present in requirements.txt and we have a version specified, replace it with the specified version
- If it's present in requirements.txt and we do not have a version specified, leave the requirements.txt entry alone

Reply:
Good logic. Can you add a comment with the explanation?